pkg/tenant/db/tenants.go

```diff
@@ -30,18 +31,17 @@ func (t *Tenant) Namespace() string {
 }
 
 func (t *Tenant) MarshalValue() ([]byte, error) {
-	// TODO: look into bson, msgpack, etc.
-	return json.Marshal(t)
+	return msgpack.Marshal(t)
 }
 
 func (t *Tenant) UnmarshalValue(data []byte) error {
-	// TODO: look into bson, msgpack, etc.
-	return json.Unmarshal(data, t)
+	return msgpack.Unmarshal(data, t)
 }
 
 func CreateTenant(ctx context.Context, tenant *Tenant) (err error) {
-	if tenant.ID == uuid.Nil {
-		tenant.ID = uuid.New()
+	if tenant.ID.Compare(ulid.ULID{}) == 0 {
```

Comment:
When we add docstrings to these methods (and we should) we should note that if the User passes in an ID then it will be used; otherwise a new ID will be created.